src/metabase/api/user.clj

```diff
@@ -151,11 +153,24 @@
      :offset offset-paging/*offset*}))
 
 
+(defn- with-general-permissions
```

Comment:
I didn't go with `hydrate` keys because these are UI-specific flags. Also,by doing this at API we could use all the bindings, so that we don't have to make the API calls.